src/Workspaces/CSharp/Portable/Extensions/CastExpressionSyntaxExtensions.cs

```diff
@@ -360,9 +360,9 @@ public static bool IsUnnecessaryCast(this CastExpressionSyntax cast, SemanticMod
                 // Explicit reference conversions can cause an exception or data loss, hence can never be removed.
                 return false;
             }
-            else if (expressionToCastType.IsExplicit && expressionToCastType.IsNumeric && IsInExplicitCheckedOrUncheckedContext(cast))
+            else if (expressionToCastType.IsExplicit && expressionToCastType.IsNumeric)
             {
-                // Don't remove any explicit numeric casts in explicit checked/unchecked context.
+                // Don't remove any explicit numeric casts.
                 // https://github.com/dotnet/roslyn/issues/2987 tracks improving on this conservative approach.
```

Comment:
I think it's still valid for the moment. #2987 is tracking an effort to be *less* conservative. It might be that the issue isn't relevant anymore, but I chose to just leave the comment for now.